### ISSUE TITLE
Flip env(safe-area-inset-*)

### ISSUE THREAD

**Issue by nex3** [MohammadYounes/rtlcss#231]
I'd expect RTLCSS to flip `env(safe-area-inset-left)` to `env(safe-area-inset-right)`, and vice versa. While these are likely to be the same on most devices, that's not guaranteed, so being able to flip them could be important for broad device compatibility.

**Comment by MohammadYounes** [comment]
In case of four-value syntax, functions are already flipped, for example:

* Input
  ```css
  body {
    padding:
      env(safe-area-inset-top, 20px)
      env(safe-area-inset-right, 20px)
      env(safe-area-inset-bottom, 20px)
      env(SAFE-AREA-INSET-LEFT, 50px)
    ;
  }
  ```
* Output
  ```css
  body {
    padding:
      env(safe-area-inset-top, 20px)
      env(SAFE-AREA-INSET-LEFT, 50px)
      env(safe-area-inset-bottom, 20px)
      env(safe-area-inset-right, 20px)
    ;
  }
  ```

In case of single value, most likely the declaration is flipped, for example:

* Input 
  ```css
  body {
   margin-left: env(safe-area-inset-left, 20px);
  }
  ```
* Output
  ```css
  body {
   margin-right: env(safe-area-inset-left, 20px);
  }
  ```

In case it was neither, users can explicitly specify the needed output, for example:

* Input
  ```css
  body {
   margin-top: /*rtl:env(safe-area-inset-left, 20px)*/env(safe-area-inset-right, 20px);
  }
  ```

* Output
  ```css
  body {
   margin-top: env(safe-area-inset-left, 20px);
  }
  ```

Based on the above, Where do you expect these to be flipped ?

**Comment by nex3** [comment]
I'd expect it to be flipped in both the first and second examples. I'd expect the outputs to be, respectively:

```css
body {
  padding:
    env(safe-area-inset-top, 20px)
    env(SAFE-AREA-INSET-RIGHT, 50px)
    env(safe-area-inset-bottom, 20px)
    env(safe-area-inset-left, 20px)
  ;
}
```

and

```css
body {
 margin-right: env(safe-area-inset-right, 20px);
}
```

Otherwise, if the device has asymmetrical safe areas, the padding/margin is no longer correctly ensuring that the content is within them.

**Comment by MohammadYounes** [comment]
I couldn't find anything stating that these variables are direction sensitive. Thus, we should treat them as any other variable.

Do browsers flip the value of  `safe-area-inset-left`/`safe-area-inset-right` based on document direction ?

**Comment by nex3** [comment]
> Do browsers flip the value of `safe-area-inset-left`/`safe-area-inset-right` based on document direction ?

I don't believe they do... which I think suggests that RTLCSS *should* flip them. Imagine a device whose screen is shaped like this:

![safe-area-inset-example](https://user-images.githubusercontent.com/188/123182435-47a82a80-d444-11eb-81a8-c16fe404a671.png)

Because the left side of the screen has curved corners, `env(safe-area-inset-left)` is `10px`. Because the right side does not, `env(safe-area-inset-right)` is `0px`. A user writing styles for an LTR language wants to ensure that their text doesn't get cut off by the corners, so they write `p {margin-left: env(safe-area-inset-left)}` which is resolved to `p {margin-left: 10px}` at render-time. Perfect!

Then that user runs RTLCSS on their stylesheet to get a new version to use with an RTL language. If it produces `p {margin-right: env(safe-area-inset-left)}`, that's resolved to `p {margin-right: 10px}`, which means that their text is hovering 10px away from the right side of the screen even though the right side isn't rounded. That's bad! What they actually want is `p {margin-right: env(safe-area-inset-right)}`, which will correctly resolve to `p {margin-right: 0px}` for this device.

**Comment by MohammadYounes** [comment]
Added to https://github.com/MohammadYounes/rtlcss/tree/feat-process-env-variables. Kindly check and let me know if you have any comments before releasing this feature.

**Comment by nex3** [comment]
Looks great, thank you!

**Comment by MohammadYounes** [comment]
Released in v3.3.0
